test/tls_test.go

```diff
@@ -1128,3 +1129,137 @@ func TestTLSHandshakeFailureMemUsage(t *testing.T) {
 		})
 	}
 }
+
+func TestTLSClientAuthWithRDNSequence(t *testing.T) {
+	for _, test := range []struct {
+		name   string
+		config string
+		certs  nats.Option
+		err    error
+		rerr   error
+	}{
+		{
+			"connect with tls using full RDN sequence",
+			`
+				port: -1
+				%s
+
+				authorization {
+				  users = [
+				    { user = "CN=localhost,OU=NATS,O=NATS,L=Los Angeles,ST=California,C=US,DC=foo1,DC=foo2" }
```

Comment:
So the issue was with two DC entries not showing up properly?

Reply:
None of the DC parts would be included the [current way that the Go stdlib](https://github.com/golang/go/blob/202c43b2ad3fca2cdcaff0d0720de5c99030b638/src/crypto/x509/pkix/pkix.go#L226) generates this string so it would become 

```
CN=localhost,OU=NATS,O=NATS,L=Los Angeles,ST=California,C=US
```

instead of

```
CN=localhost,OU=NATS,O=NATS,L=Los Angeles,ST=California,C=US,DC=foo1,DC=foo2
```